test/boost/managed_bytes_test.cc

```diff
@@ -325,3 +326,29 @@ SEASTAR_THREAD_TEST_CASE(test_managed_bytes_view_substr) {
     mvs = mv.substr(offset, len);
     BOOST_REQUIRE_EQUAL(mvs, v);
 }
+
+SEASTAR_THREAD_TEST_CASE(test_managed_bytes_view_appending_hash) {
+    bytes b;
+    auto hseed = tests::random::get_int<uint64_t>();
+    auto h1 = xx_hasher(hseed);
+    auto h2 = xx_hasher(hseed);
+    managed_bytes_view mv;
+
+    BOOST_REQUIRE_EQUAL(h1.finalize(), h2.finalize());
+
+    appending_hash<bytes>{}(h1, b);
+    appending_hash<managed_bytes_view>{}(h2, mv);
+    BOOST_REQUIRE_EQUAL(h1.finalize(), h2.finalize());
+
+    b = random_bytes(0, 16);
+    mv = managed_bytes_view(b);
+    appending_hash<bytes>{}(h1, b);
+    appending_hash<managed_bytes_view>{}(h2, mv);
+    BOOST_REQUIRE_EQUAL(h1.finalize(), h2.finalize());
+
+    b = random_bytes(16, 131072);
+    mv = managed_bytes_view(b);
+    appending_hash<bytes>{}(h1, b);
+    appending_hash<managed_bytes_view>{}(h2, mv);
+    BOOST_REQUIRE_EQUAL(h1.finalize(), h2.finalize());
+}
```

Comment:
What does this test? In all cases the managed_bytes will not be fragmented.

Reply:
Correct, I'll change to convert via `managed_bytes`.
Is there a way to enforce  fragmentation of `managed_bytes`? (command line option maybe?)

Reply:
No. I suggest adding `static managed_bytes managed_bytes::with_fragment_sizes(std::vector<size_t>)` for tests.

Reply:
I'll add auto-fragmentation during normal operation, but the API above can still be useful for tests.

Reply:
@avikivity maybe just support passing an optional allocation function to the constructor.
Then in the test we can use it to control the allocated fragment sizes whichever way we like.